modules/dag-l1/src/main/scala/org/tessellation/dag/l1/modules/Programs.scala

```diff
@@ -1,22 +1,34 @@
 package org.tessellation.dag.l1.modules
 
-import cats.effect.Sync
+import cats.effect.Async
+import cats.effect.std.Random
 
+import org.tessellation.dag.l1.domain.snapshot.programs.GlobalSnapshotPulling
 import org.tessellation.dag.l1.http.p2p.P2PClient
 import org.tessellation.sdk.domain.cluster.programs.{Joining, L0PeerDiscovery, PeerDiscovery}
 import org.tessellation.sdk.modules.SdkPrograms
 
 object Programs {
 
-  def make[F[_]: Sync](sdkPrograms: SdkPrograms[F], p2pClient: P2PClient[F], storages: Storages[F]): Programs[F] = {
+  def make[F[_]: Async: Random](
+    sdkPrograms: SdkPrograms[F],
+    p2pClient: P2PClient[F],
+    storages: Storages[F]
+  ): Programs[F] = {
     val l0PeerDiscovery = L0PeerDiscovery.make(p2pClient.l0Cluster, storages.l0Cluster)
+    val globalSnapshotPulling = GlobalSnapshotPulling.make(
```

Comment:
To be honest, I'd use L0 as an external service/API, hence encapsulate all L0-related things into a separate service

Reply:
Not sure if the way it is used in `StateChannel` class makes it a service 🤔 .

Reply:
what do you mean? 🤔

Reply:
I meant to gather all L0-related things into a single "facade" and treat it as an external service